src/cut.rs

```diff
@@ -26,10 +26,10 @@ pub fn cut_out_bytes(
     let mut out_buf = BytesMut::new();
 
     let mut remaining = size;
-    loop {
-        let mut fragment = vec![0; fragment_size-1];
```

Comment:
I kept it to ensure `fragment` has the correct length when it gets filled by `to_vec()`. Without it, `to_vec()` writes the full `content` into `fragment`. Anyway, it's not a good implementation.